src/plots/plots.js

```diff
@@ -2406,6 +2407,11 @@ plots.doCalcdata = function(gd, traces) {
         calcdata[i] = cd;
     }
 
+    // 'regular' loop - make sure container traces (eg carpet) calc before
+    // contained traces (eg contourcarpet)
+    for(i = 0; i < fullData.length; i++) calci(i, true);
+    for(i = 0; i < fullData.length; i++) calci(i, false);
```

Comment:
Because I moved `convertColumnData` and `setConvert` for `carpet` traces from the `supplyDefaults` step to the `calc` step, we need to ensure `carpet.calc` happens before `contourcarpet.calc` and `scattercarpet.calc`. I'm not sure if we're ever going to have another situation like this where one trace essentially lives inside another, but in case we do this seemed like a reasonably generic way to handle it.